### PR TITLE
fix(cmd): close adapters on shutdown; propagate writeJSON errors; add edge-case tests

### PR DESCRIPTION
**Pull request by sergeyklay**
### 🎯 Scope & Context

**Type:** Fix

**Intent:** Address Minor findings from the `cmd/sortie/main.go` code review: optional adapter teardown, idiomatic error checking, propagated `writeJSON` errors, and the previously uncovered edge-case test paths.

**Related Issues:** #265

### 🧭 Reviewer Guide

**Complexity:** Low

#### Entry Point

`cmd/sortie/main.go` — three independent improvements applied after the tracker and agent adapters are constructed: optional `io.Closer` teardown (lines ~219 and ~233), `errors.Is` in the HTTP server goroutine, and `writeJSON` now returns an `error` so callers can act on it. Start there, then cross-reference `closer_test.go` for the spy infrastructure.

#### Sensitive Areas

- `cmd/sortie/main.go`: The `io.Closer` defers are best-effort (`//nolint:errcheck`); a failing `Close()` is logged nowhere — this is intentional parity with the existing HTTP-server shutdown path.
- `cmd/sortie/main.go`: `serverShutdownTimeout` is a package-level variable mutated only in tests; it must never be set to zero in production code.
- `cmd/sortie/closer_test.go`: Uses `sync.Once` + `sync.Map` to register test-only adapter kinds exactly once across parallel tests. The `closeableOnce` guard prevents registry panic on duplicate registration.

### ⚠️ Risk Assessment

- **Breaking Changes:** No breaking changes
- **Migrations/State:** No migrations or state changes

### CODE REVIEW

File: cmd/sortie/main.go [hunk 1]
```diff
@@ -351,7 +362,7 @@ func run(ctx context.Context, args []string, stdout io.Writer, stderr io.Writer)
 	o.Run(ctx)
 
 	if srv != nil {
-		shutdownCtx, shutdownCancel := context.WithTimeout(context.Background(), 5*time.Second)
+		shutdownCtx, shutdownCancel := context.WithTimeout(context.Background(), serverShutdownTimeout)
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply cmd/sortie/main.go:365]: The test that mutates this variable (`TestRunServerShutdownError`) explicitly opts out of parallelism with `// No t.Parallel: mutates package-level serverShutdownTimeout`. Go's test runner serializes non-parallel tests within a package, so no concurrent read/write is possible. The suite passes with `-race` cleanly.

Injecting via an options struct would add a parameter to `run()` used only by one test, complicating the production API for a non-existent race. Keeping this as-is.

File: cmd/sortie/main_test.go [hunk 2]
```diff
@@ -1508,3 +1519,266 @@ func TestValidateDoesNotStartWatcher(t *testing.T) {
...
+	}
+	if !strings.Contains(subStderr.String(), "sortie started") {
+		cmd.Process.Kill() //nolint:errcheck // best-effort cleanup
+		t.Fatalf("subprocess did not reach 'sortie started' within 5 s; stderr:\n%s", subStderr.String())
+	}
+
+	// Send SIGINT — should trigger context cancellation and clean shutdown.
+	if err := cmd.Process.Signal(syscall.SIGINT); err != nil {
+		t.Fatalf("Signal(SIGINT): %v", err)
+	}
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply cmd/sortie/main_test.go:1644]: `syscall.SIGINT` maps to `os.Interrupt` on Windows, and `Process.Signal(os.Interrupt)` works cross-platform since Go 1.14+. The helper-subprocess pattern compiles and functions identically on all three target platforms (linux, darwin, windows). No existing tests in the repo use `//go:build` tags for signal handling. Keeping as-is.

File: cmd/sortie/main_test.go [hunk 3]
```diff
@@ -1508,3 +1519,266 @@ func TestValidateDoesNotStartWatcher(t *testing.T) {
...
+		t.Errorf("stderr = %q, want to contain %q", stderr.String(), "http server shutdown error")
+	}
+}
+
+// TestRunReadOnlyWorkflowDir covers the persistence.Open error path that
+// fires when the database file cannot be created because the workflow
+// directory has no write permission.
+func TestRunReadOnlyWorkflowDir(t *testing.T) {
+	// No t.Parallel: calls t.Chdir via setupRunDir, and mutates permissions.
+	if os.Getuid() == 0 {
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply cmd/sortie/main_test.go:1754]: `os.Getuid()` compiles on all platforms — on Windows it returns `-1`, so the skip condition never triggers, which is correct since Windows lacks Unix permission semantics for this test. The suggested `os.Getenv("USER") == "root"` is unreliable: `$USER` can be unset, overridden, or absent in non-interactive environments (containers, CI). `os.Getuid() == 0` is the canonical Go pattern, used in 4 other places in this codebase (`exit_test.go`, `preflight_test.go` x3). Keeping for consistency.

File: cmd/sortie/closer_test.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+	writeIssuesFixture(t, dir)
+
+	token, trackerFlag, agentFlag := newCloseToken(t)
+	wfPath := writeCloseableWorkflow(t, dir, "issues.json", token)
+
+	var stdout, stderr bytes.Buffer
+	ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond)
+	defer cancel()
+
+	code := run(ctx, []string{wfPath}, &stdout, &stderr)
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply cmd/sortie/closer_test.go:170]: 500ms `context.WithTimeout` is the established convention across 9 instances in `cmd/sortie/*_test.go` (including `QuickStartScenario`, `TestRunPrecancelledContext`, etc.), all passing reliably with `-race`. The mock adapter returns instantly — no real I/O, network, or subprocess is involved — so 500ms provides ample headroom. Changing one instance would break consistency with the rest of the suite.
